reference_guide/navigation/navigation_intro.md

```diff
@@ -0,0 +1,93 @@
+in the appropriate editing or display window, and requesting focus. For example:
+* Opening [Files, classes or symbols](https://www.jetbrains.com/help/idea/navigating-to-class-file-or-symbol-by-name.html),
+  in an editor. This also applies to editing content in preference windows, such as 
+  [Copyright profiles](https://www.jetbrains.com/help/idea/copyright-profiles.html).
+* Going to locations within files based on [Bookmarks](https://www.jetbrains.com/help/idea/navigating-with-bookmarks.html),
+  [TODOs](https://www.jetbrains.com/help/idea/using-todo.html), 
+  or [Lines](https://www.jetbrains.com/help/idea/2017.3/navigating-to-line.html). 
+
+### The Navigatable Interface 
+The [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface defines the following methods of particular interest:
+
+```java
+
+ boolean canNavigate();
+
+ boolean canNavigateToSource();
+ 
+ void navigate(boolean requestFocus); 
+```
+The `canNavigate()` method indicates whether this object supports navigation of any kind. The `canNavigateToSource()` method
+indicates whether this object supports navigation to source within an editor. See the 
+[Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface comments for more information about these two methods.
+
+The `navigate()` method is called to open the (content-containing) object in an editor or display, and request focus. Typically an implementation
+of `Navigatable` does the heavy lifting of invoking an editor for the content and requesting focus. See below for examples,
+and the [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface for more information.  
+
+### Example IntelliJ Platform Implementations of Navigatable
+This section describes some example implementations that can be found in the IntelliJ Platform. This is by no means an exhaustive list, 
+but is intended to illustrate how `Navigatable` works in different contexts.
+
+#### Example Navigatable Class Related to Psi Elements
+
+The [NavigatablePsiElement](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/psi/NavigatablePsiElement.java) 
+interface is the basis for many implementation classes, abstract classes, and interfaces relating to `Navigatable` 
+Psi elements. The class diagram below shows `NavigatablePsiElement` inherits from three interfaces defining the underlying
+functionality of a `Navigatable` Psi element.
+
+<img src="img/navigatable_psi_element.png" alt="NavigatablePsiElement" width="500px"/>
+
+For example, the subclass [JavaStubPsiElement](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/JavaStubPsiElement.java) 
```

Comment:
This is true but irrelevant. Plugin developers almost never need to care about `JavaStubPsiElement`.